custom_components/homewhiz/sensor.py

```diff
@@ -1,204 +1,71 @@
-        minutes = (
-            clamp(data[self._progress.minute.wifiArrayIndex])
-            if self._progress.minute is not None
-            else 0
-        )
-        return hours * 60 + minutes
-
-
-def generate_sensor_descriptions_from_features(features: list[ApplianceFeature]):
-    result = []
-    for feature in features:
-        read_index = feature.wifiArrayIndex
-        if feature.boundedValues is not None:
-            for bounds in feature.boundedValues:
-                result.append(
-                    SubProgramBoundedSensorEntityDescription(
-                        feature.strKey, bounds, read_index
-                    )
-                )
-    return result
-
-
-def generate_sensor_descriptions_from_config(
-    config: ApplianceConfiguration,
-) -> list[HomeWhizSensorEntityDescription]:
-    _LOGGER.debug("Generating descriptions from config")
-    result = []
-    if config.deviceSubStates is not None:
-        _LOGGER.debug("Adding SUB_STATE EnumEntityDescription")
-        result.append(
-            EnumSensorEntityDescription(
-                "SUB_STATE",
-                config.deviceSubStates.subStates,
-                config.deviceSubStates.wifiArrayReadIndex,
-            )
-        )
-    result.extend(generate_sensor_descriptions_from_features(config.subPrograms))
-    if config.progressVariables is not None:
-        _LOGGER.debug("Adding config progress variables")
-        for field in fields(config.progressVariables):
-            feature = getattr(config.progressVariables, field.name)
-            if feature is not None:
-                result.append(
-                    ProgressSensorEntityDescription(feature),
-                )
-    if config.monitorings is not None:
-        _LOGGER.debug("Adding config monitorings")
-        result.extend(generate_sensor_descriptions_from_features(config.monitorings))
-
-    return result
-
-
-class HomeWhizSensorEntity(CoordinatorEntity[HomewhizCoordinator], SensorEntity):
-    _attr_has_entity_name = True
-
+class HomeWhizSensorEntity(HomeWhizEntity, SensorEntity):
     def __init__(
         self,
         coordinator: HomewhizCoordinator,
-        description: HomeWhizSensorEntityDescription,
-        entry: ConfigEntry,
+        control: TimeControl | EnumControl | NumericControl,
+        device_name: str,
         data: EntryData,
     ):
-        super().__init__(coordinator)
-        unique_name = entry.title
-        self._attr_unique_id = f"{unique_name}_{description.key}"
-        self._attr_device_info = build_device_info(unique_name, data)
-
-        self._localization = data.contents.localization
-        self.entity_description = description
-        self._value_fn = description.value_fn
+        super().__init__(coordinator, device_name, control.key, data)
+        self._control = control
+        if isinstance(control, TimeControl):
+            self._attr_icon = "mdi:clock-outline"
+            self._attr_native_unit_of_measurement = "min"
```

Comment:
Should be maybe use units from const?
https://github.com/home-assistant/core/blob/dev/homeassistant/const.py#L592

Reply:
Good idea, but it has not yet been released: https://github.com/home-assistant/core/commit/6295cf3dc4d017e6f3151c2b8014e26bada78343